modules/spring-config/src/main/java/org/mule/runtime/config/spring/dsl/api/AttributeDefinition.java

```diff
@@ -0,0 +1,240 @@
+    {
+        if (configParameterName != null)
+        {
+            visitor.onConfigurationParameter(configParameterName, defaultValue);
+        }
+        else if (referenceObject != null)
+        {
+            visitor.onReferenceObject(referenceObject);
+        }
+        else if (hasDefaultValue)
+        {
+            visitor.onFixedValue(defaultValue);
+        }
+        else if (undefinedSimpleParametersHolder)
+        {
+            visitor.onUndefinedSimpleParameters();
+        }
+        else if (undefinedComplexParametersHolder)
+        {
+            visitor.onUndefinedComplexParameters();
+        }
+        else if (referenceSimpleParameter != null)
+        {
+            visitor.onReferenceSimpleParameter(referenceSimpleParameter);
+        }
+        else if (childObjectType != null && collection)
+        {
+            visitor.onComplexChildList(childObjectType);
+        }
+        else if (childObjectType != null)
+        {
+            visitor.onComplexChild(childObjectType);
+        }
+        else if (valueFromTextContent)
+        {
+            visitor.onValueFromTextContent();
+        }
+        else
+        {
+            throw new RuntimeException();
+        }
+    }
+
+    public static class Builder
+    {
+
+        private AttributeDefinition attributeDefinition = new AttributeDefinition();
+
+        private Builder()
+        {
+        }
+
+        /**
+         * @param configParameterName name of the configuration parameter from which this attribute value will be extracted.
+         * @return the builder
+         */
+        public static Builder fromSimpleParameter(String configParameterName)
+        {
+            Builder builder = new Builder();
+            builder.attributeDefinition.configParameterName = configParameterName;
+            return builder;
+        }
+
+        /**
+         * @param defaultValue defines the default value to be used for the attribute if no other value is provided.
+         * @return the builder
+         */
+        public Builder withDefaultValue(Object defaultValue)
+        {
+            attributeDefinition.hasDefaultValue = true;
+            attributeDefinition.defaultValue = defaultValue;
+            return this;
+        }
+
+        /**
+         * @param value a fixed value which will be assigned to the attribute.
+         * @return the builder
+         */
+        public static Builder fromFixedValue(Object value)
+        {
+            Builder builder = new Builder();
+            builder.attributeDefinition.hasDefaultValue = true;
+            builder.attributeDefinition.defaultValue = value;
+            return builder;
+        }
+
+        /**
+         * Calling this method declares that the attribute will be assigned with all declared
+         * simple configuration attribute and its value. By simple attribute we consider those
+         * with a key and a string value as content.
```

Comment:
For now yes since we just get the value from configuration and send it as it is. This may be updated in the future based on other needs.